drivers/hmis/spec/models/hmis/ce/candidate_pool_spec.rb

```diff
@@ -6,29 +6,33 @@
   let!(:ds1) { create :hmis_data_source }
   let!(:p1) { create :hmis_hud_project, data_source: ds1, project_type: 13 }
   let!(:project_config) { create :hmis_project_ce_config, supports_waitlist_referrals: true, project: p1 }
+  # Initial setup: project with 1 unit group with 2 units (no opportunities/candidate pools yet)
```

Comment:
@martha I _think_ this update keeps the test logic in place, but it could use another pair of eyes.

The tests like `.active_for_maintenance with open opportunities` are testing against `opp` which references a candidate pool directly, and belongs to a unit group that does _not_ reference a candidate pool. I am pretty sure that matches the old behavior.

Tests that redefine `let!(:ug)` are the only ones that assign a candidate pool to the unit group.

Reply:
Looks good, I think your changes preserve the previous logic. I might propose an update to the the setup though, now that I am looking at this again. I think the the normal/default scenario we are testing for is "unit group associated with a non-stale candidate pool," so what if we make the initial setup reflect the normal scenario, and then the additional contexts modify from there, to be more like real life? I played around with this locally and pushed up a proposal, https://github.com/greenriver/hmis-warehouse/pull/6084/changes

Let me know what you think. I'm also happy to merge what you have as-is.